src/main/java/courgette/runtime/CourgettePickleMatcher.java

```diff
@@ -26,7 +26,7 @@ public boolean matches() {
 
         try {
             compiler.compile(cucumberFeature.getGherkinFeature()).forEach(pickle -> {
-                matched.set(filters.matchesFilters(new PickleEvent(cucumberFeature.getUri(), pickle)));
+                matched.set(filters.matchesFilters(new PickleEvent(cucumberFeature.getUri().toString(), pickle)));
```

Comment:
this needs to use the scheme specific part of the URI. If we use toString() it would prefix the path with file://

```java
cucumberFeature.getUri().getSchemeSpecificPart()
````